packages/codemods/src/transforms/transform-grid-api-methods-v31/replacements.ts

```diff
@@ -7,106 +7,141 @@ import {
-      api: p.expression(),
-      value: p.expression(),
-    }),
-    template(
-      ({ api, value }) =>
-        ast.expression`${api}.setGridOption('getServerSideGroupLevelParams', ${value})`,
-    ),
-  ),
+  ...['', '?', '!']
+    .map((apiCoalesce) => [
+      replace(
+        matchNode(({ api }) => ast.expression`${api}${apiCoalesce}.refreshServerSideStore()`, {
+          api: p.expression(),
+        }),
+        template(({ api }) => ast.expression`${api}${apiCoalesce}.refreshServerSide()`),
+      ),
+      replace(
+        matchNode(
+          ({ api, params }) =>
+            ast.expression`${api}${apiCoalesce}.refreshServerSideStore(${params})`,
+          {
+            api: p.expression(),
+            params: p.expression(),
+          },
+        ),
+        template(
+          ({ api, params }) => ast.expression`${api}${apiCoalesce}.refreshServerSide(${params})`,
+        ),
+      ),
+      replace(
+        matchNode(({ api }) => ast.expression`${api}${apiCoalesce}.getServerSideStoreState()`, {
+          api: p.expression(),
+        }),
+        template(({ api }) => ast.expression`${api}${apiCoalesce}.getServerSideGroupLevelState()`),
+      ),
+      replace(
+        matchNode(
+          ({ api, value }) =>
+            ast.expression`${api}${apiCoalesce}.setProcessSecondaryColDef(${value})`,
+          {
+            api: p.expression(),
+            value: p.expression(),
+          },
+        ),
+        template(
+          ({ api, value }) =>
+            ast.expression`${api}${apiCoalesce}.setGridOption('processPivotResultColDef', ${value})`,
+        ),
+      ),
+      replace(
+        matchNode(
+          ({ api, value }) =>
+            ast.expression`${api}${apiCoalesce}.setProcessSecondaryColGroupDef(${value})`,
+          {
+            api: p.expression(),
+            value: p.expression(),
+          },
+        ),
+        template(
+          ({ api, value }) =>
+            ast.expression`${api}${apiCoalesce}.setGridOption('processPivotResultColGroupDef', ${value})`,
+        ),
+      ),
+      replace(
+        matchNode(
+          ({ api, value }) =>
+            ast.expression`${api}${apiCoalesce}.setGetServerSideStoreParams(${value})`,
+          {
+            api: p.expression(),
+            value: p.expression(),
+          },
+        ),
+        template(
+          ({ api, value }) =>
+            ast.expression`${api}${apiCoalesce}.setGridOption('getServerSideGroupLevelParams', ${value})`,
+        ),
+      ),
+    ])
+    .flat(),
 ];
 
 const GRID_API_DEPRECATIONS: Array<GridApiDeprecation> = [
-  matchNode(({ api }) => ast.expression`${api}.setGetRowId()`, {
-    api: p.expression(),
-  }),
-];
-
-const COLUMNS_API_REPLACEMENTS: Array<GridApiReplacement> = [
-  replace(
-    matchNode(({ api }) => ast.expression`${api}.getAllColumns()`, {
+  ...['', '?', '!'].map((apiCoalesce) =>
```

Comment:
Do you think we should push this up and change the format of the config to return a function(apiCoalesce) => array? Save repeating that in every generator?

Reply:
Well yes I thought about it but looked much more complex than this - i will give it another try.
There is also the problem that we might run multiple times transformations that don't use the apiCoalesce (that I will rename by the way)

Reply:
Ok if this makes it more complex maybe lets not worry about that.

Reply:
We agreed that the number of changes required to make this work would be too big - for now we can live with this and consider a bigger refactoring in the future. The advantage of this is, whilst is more verbose, is very clear what is happening.